src/dhcpmon/src/dhcp_device.cpp

```diff
@@ -61,36 +62,37 @@ std::shared_ptr<swss::Table> mStateDbMuxTablePtr = std::make_shared<swss::Table>
 /* interface to vlan mapping */
 std::unordered_map<std::string, std::string> vlan_map;
 
+/* list of portchannels */
+std::vector<std::string> portchannel_list;
+
 /** Berkeley Packet Filter program for "udp and (port 67 or port 68)".
  * This program is obtained using the following command tcpdump:
- * `tcpdump -dd "outbound and udp and (port 67 or port 68)"`
+ * `tcpdump -dd "udp and (port 67 or port 68)"`
  */
```

Comment:
Hi Kelly, can we still use origin bpf filter, and follow below logic ?

Inbound listen all interfaces
1. int is vlan member, update Ethernet Discover/Request RX
2. int is vlan, update Vlan Discover/Request RX
3. int is port-channel, update port-channel Offer/Ack RX

Outbond  listen all interfaces
1. int is vlan member, update Ethernet Offer/Ack TX
2. int is vlan, update Vlan Offer/Ack Tx
3. int is port-channel. update port-channel Discover/Request TX

Vlan AGG Discover RX = Vlan Discover RX, TX = port-channel Discover TX
Vlan AGG Offer RX = port-channel Offer RX, TX = Ethernet Offer TX
Vlan AGG Request RX = Vlan Request RX, TX = port-channel Request TX
Vlan AGG Ack RX = port-channel Ack RX, TX = Ethernet Ack TX

Reply:
Hi Jie, what you proposed is the logic we have right now before this PR(with a bug in offer/ack rx), but the agg counter is different from the original dhcpmon counter.

Original:
[ eth0- Current rx/tx] Discover: 0/ 0, Offer: 0/ 0, Request: 0/ 0, ACK: 0/ 0
[ eth0- Current rx/tx] Discover: 0/ 0, Offer: 0/ 0, Request: 0/ 0, ACK: 0/ 0
[ PortChannel104- Current rx/tx] Discover: 0/ 1, Offer: 0/ 0, Request: 0/ 3, ACK: 0/ 0
[ PortChannel103- Current rx/tx] Discover: 0/ 1, Offer: 0/ 0, Request: 0/ 3, ACK: 0/ 0
[ PortChannel102- Current rx/tx] Discover: 0/ 2, Offer: 1/ 0, Request: 0/ 6, ACK: 1/ 0
[ PortChannel101- Current rx/tx] Discover: 0/ 0, Offer: 0/ 0, Request: 0/ 0, ACK: 0/ 0
[ Vlan1000- Current rx/tx] Discover: 1/ 0, Offer: 0/ 1, Request: 3/ 0, ACK: 0/ 1
[ Agg-Vlan1000- Current rx/tx] Discover: 1/ 4, **Offer: 1/ 1,** Request: 3/ 12, **ACK: 1/ 1**

Current:
[ eth0- Current rx/tx] Discover: 0/ 0, Offer: 0/ 0, Request: 0/ 0, ACK: 0/ 0
[ eth0- Current rx/tx] Discover: 0/ 0, Offer: 0/ 0, Request: 0/ 0, ACK: 0/ 0
[ PortChannel104- Current rx/tx] Discover: 0/ 1, Offer: 1/ 0, Request: 0/ 3, ACK: 1/ 0
[ PortChannel103- Current rx/tx] Discover: 0/ 1, Offer: 1/ 0, Request: 0/ 3, ACK: 1/ 0
[ PortChannel102- Current rx/tx] Discover: 0/ 2, Offer: 1/ 0, Request: 0/ 6, ACK: 1/ 0
[ PortChannel101- Current rx/tx] Discover: 0/ 0, Offer: 1/ 0, Request: 0/ 0, ACK: 1/ 0
[ Vlan1000- Current rx/tx] Discover: 1/ 0, Offer: 0/ 1, Request: 3/ 0, ACK: 0/ 1
[ Agg-Vlan1000- Current rx/tx] Discover: 1/ 4, **Offer: 4/ 1,** Request: 3/ 12, **ACK: 4/ 1**

I wanted to keep the agg counter consistent by changing:
Vlan AGG Offer/Ack RX = port-channel Offer/Ack RX
to:
Vlan AGG Offer/Ack RX = Vlan Offer/Ack RX
And keeping track of Offer/Ack RX on Portchannels but not adding them to agg-counter.

Though according to the this function of checking if dhcp relay message is healthy here https://github.com/sonic-net/sonic-buildimage/blob/f8a393c3a176d120bc6472f3b56936ef85916d4a/src/dhcpmon/src/dhcp_device.cpp#L368, the new logic would work just fine.
There's no official documentation of dhcpmon counter. We could potentially change it as we like as long as the disparity alarm works fine